v2/yarpcerror/errors.go

```diff
@@ -25,31 +25,52 @@ import (
 	"fmt"
 )
 
-// Newf returns a new Status.
+// Newf returns a new Status, with message formatting.
 //
 // The Code should never be CodeOK, if it is, this will return nil.
-func Newf(code Code, format string, args ...interface{}) *Status {
+func Newf(code Code, format string, args ...interface{}) error {
 	if code == CodeOK {
 		return nil
 	}
-	return &Status{
+	return New(code, sprintf(format, args...))
+}
+
+// New return a new Status.
+//
+// The Code should never be CodeOK; if it is, this will return nil.
+func New(code Code, message string, options ...StatusOption) error {
+	if code == CodeOK {
+		return nil
+	}
+	status := &status{
 		code:    code,
-		message: sprintf(format, args...),
+		message: message,
 	}
+	for _, opt := range options {
+		opt.apply(status)
+	}
+
+	// The name must only contain lowercase letters from a-z and dashes (-), and
```

Comment:
nit: can we move this documentation to the function declaration for `validateName`?